### DISCUSSION TITLE
2006/01/24/vc%E4%B8%AD%E7%9A%84%E5%AD%97%E8%8A%82%E5%AF%B9%E9%BD%90

### DISCUSSION THREAD

**Discussion by giscus** [ruanqizhen/blog#73] [Announcements]
# 2006/01/24/vc%E4%B8%AD%E7%9A%84%E5%AD%97%E8%8A%82%E5%AF%B9%E9%BD%90

当在C中定义了一个结构类型时，它的大小是否等于各字段(field)大小之和？编译器将如何在内存中放置这些字段？ANSI C对结构体的内存布局有什么要求？而我们的程序又能否依赖这种布局？这些问题或许对不少朋友来说还有点模糊，那么本文就试着探究它们背后的秘密。

https://blog.qizhen.xyz/2006/01/24/vc%E4%B8%AD%E7%9A%84%E5%AD%97%E8%8A%82%E5%AF%B9%E9%BD%90

**Comment by ruanqizhen** [comment]
之前博客下的留言：

禹
2007年11月1日 11:20 编辑
高手 谢谢告诉我内存对齐的用法 我是通过Google搜到你的spaces

回复
Leona♡
2006年01月26日 16:10 编辑
这是什么这是什么 !!!!!!!!!!!!!!!!完全不明白啊~~~~~~~

回复
Christy
2006年01月26日 02:23 编辑
沙發~有空到我的blog做客新年快樂~
